### PR TITLE
WINC-674: [vendor] move to openshift/api

### PR DESCRIPTION
**Pull request by alinaryan**
Machine API resources have moved from openshift/machine-api to openshift/api.

openshift/machine-api-operator imports were replaced with openshift/api
generated client imports were replaced with client-go

### CODE REVIEW

**Review by saifshaikh48 — COMMENTED** [review]
Looks good! Suggestion to separate the code changes and the vendoring into separate commits for readability.

File: test/e2e/create_test.go [hunk 1]
```diff
@@ -193,7 +193,8 @@ func (tc *testContext) createWindowsMachineSet(replicas int32, windowsLabel bool
 	if err != nil {
 		return nil, err
 	}
-	return tc.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Create(context.TODO(), machineSet, metav1.CreateOptions{})
+	return tc.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Create(context.TODO(), machineSet,
+		metav1.CreateOptions{})
```
Comment (aravindhp) [inline test/e2e/create_test.go:197]: Please revert this change

File: go.mod [hunk 2]
```diff
@@ -2,33 +2,28 @@ module github.com/openshift/windows-machine-config-operator
 
 go 1.16
 
-replace (
```
Comment (aravindhp) [inline go.mod:5]: I suggest naming this commit `[vendor] Update openshift/api and openshift/client-go`.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @alinaryan. Just some minor change requests.

File: test/e2e/providers/vsphere/vsphere.go [hunk 3]
```diff
@@ -10,16 +10,15 @@ import (
...
+	mapi "github.com/openshift/api/machine/v1beta1"
 	core "k8s.io/api/core/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 
 	"github.com/openshift/windows-machine-config-operator/test/e2e/clusterinfo"
 )
 
-const defaultCredentialsSecretName = "vsphere-cloud-credentials"
+const defaultCredentialsSecretName = "mapi-cloud-credentials"
```
Comment (aravindhp) [inline test/e2e/providers/vsphere/vsphere.go:21]: The vSphere e2e test is [failing](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/788/pull-ci-openshift-windows-machine-config-operator-master-vsphere-e2e-operator/1455974928560951296#1:build-log.txt%3A93) because of this.

**Review by aravindhp — APPROVED** [review]
/approve

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

Thanks for working on this @alinaryan
